pgsync/sync.py

```diff
@@ -1004,10 +1009,13 @@ def poll_db(self) -> None:
                 os._exit(-1)
 
             while conn.notifies:
+                if len(item_queue)>=REDIS_WRITE_CHUNK_SIZE:
```

Comment:
I've been thinking about this and wondered if there is an opportunity to just have one bulk_push in this method. Apart from that looks good. Great job.

Reply:
Yeah, I was trying to find a way to do that. My worry is that if the length of `item_queue` exceeds `REDIS_WRITE_CHUNK_SIZE` while inside the `conn.notifies` loop, we would exceed the max size. Did you have any ideas?

Reply:
That makes sense. I was trying to get some timings against the old method to put for reference.

Reply:
I have a sample repo you can use here https://github.com/JacobReynolds/PGSyncSpeedTest that has both methods available if you want to use that for timing

Reply:
This yields significant improvements for write operations.

Slow timer 0:00:10.480808 (10.48 sec)
Fast timer 0:00:00.651109 (0.65 sec)

10K records and batch 1K. 

Nice work!